packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,33 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+///
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// In buttons that have an icon, we generally tend to use a `Row` widget
```

Comment:
avoid "we" (we use the passive voice in the API docs).

But also, I would generally avoid mentioning how this is implemented, because we might change it.

Reply:
I believe it is useful right now but I could add a todo or a note that states changing the docs of ButtonIconAlignment to be without implementation if used with other types of buttons that do not use `Row`.

So what do you think?

Reply:
I have changes as requested, now without implementation